### PR TITLE
Move KeypadContext into MathInput

### PR DESCRIPTION
**Pull request by handeyeco**
## Summary:
KeypadContext seemed to make more sense to me as part of MathInput (vs Perseus), so I thought I'd pitch this PR.

Issue: LC-1078

### CODE REVIEW

File: packages/math-input/src/components/keypad-context.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@
  */
 import * as React from "react";
 
-import type {RendererInterface} from "./types";
+import type {RendererInterface} from "@khanacademy/perseus";
```
Comment (jeremywiebe) [inline packages/math-input/src/components/keypad-context.ts:11]: 🤔  This has the unfortunate result though of creating a circular dependency (math-input depends on Perseus, Perseus depends on math-input). I'm not sure that's even allowed.

**Review by jeremywiebe — CHANGES_REQUESTED** [review]
I like the idea, but I think we need to work out the circular dependency issue.  Maybe a new interface that contains the subset of renderer methods that users of the context need?

File: packages/math-input/src/components/keypad-context.ts (hunk 1 above)
Comment (jeremywiebe) [reply packages/math-input/src/components/keypad-context.ts:11]: > @jeremywiebe I'm wondering if shared types should be in perseus-core?

That was my gut feel too. Slowly move these types that need to be shared into perseus-core.  I don't want it to become a complete dumping ground, but I think this move makes sense. 👍

Comment (handeyeco) [reply packages/math-input/src/components/keypad-context.ts]: So I moved what I needed to move into a types.ts folder in `perseus-core`. If you're still getting a bad vibe from this PR I don't mind just closing it out. Just makes sense to me that the KeypadContext should live closer to the Keypad.

**Review by jeremywiebe — APPROVED** [review]
Looks good.  I see that we'll probably start moving more shared types into core. That feels right to me.

**Pull request merged**